src/lms-harmonizer/tests_integration_pgsql/procs/harmonize_sections/test_lmssections_canvas.py

```diff
@@ -16,112 +16,113 @@
-        # arrange
-        with PgsqlConnection(test_db_config).pyodbc_conn() as connection:
-            insert_lms_section(connection, SIS_ID, SOURCE_SYSTEM)
-            insert_edfi_section(connection, SIS_ID, SECTION_ID)
-
-        # act
-        run_harmonizer(test_db_config)
-
-        # assert
-        with PgsqlConnection(test_db_config).pyodbc_conn() as connection:
-            LMSSection = query(
-                connection,
-                "select edfisectionid from lms.lmssection",
-            )
-            assert len(LMSSection) == 1
-            assert LMSSection[0]["edfisectionid"] == SECTION_ID
-
-
-def describe_when_lms_and_ods_tables_have_a_match_to_deleted_record():
-    SECTION_ID = "10000000-0000-0000-0000-000000000000"
-    SIS_ID = "sis_id"
-
-    def it_should_ignore_the_deleted_record(test_db_config: PgsqlServerConfig):
-        # arrange
-        with PgsqlConnection(test_db_config).pyodbc_conn() as connection:
-            insert_lms_section_deleted(connection, SIS_ID, SOURCE_SYSTEM)
-            insert_edfi_section(connection, SIS_ID, SECTION_ID)
-
-        # act
-        run_harmonizer(test_db_config)
-
-        # assert
-        with PgsqlConnection(test_db_config).pyodbc_conn() as connection:
-            LMSSection = query(
-                connection,
-                "select edfisectionid from lms.lmssection",
-            )
-            assert len(LMSSection) == 1
-            assert LMSSection[0]["edfisectionid"] is None
-
-
-def describe_when_lms_and_ods_tables_have_one_match_and_one_not_match():
-    SECTION_ID = "10000000-0000-0000-0000-000000000000"
-    SIS_ID = "sis_id"
-    NOT_MATCHING_SIS_ID = "not_matching_sis_id"
-
-    def it_should_run_successfully(test_db_config: PgsqlServerConfig):
-        # arrange
-        with PgsqlConnection(test_db_config).pyodbc_conn() as connection:
-            insert_lms_section(connection, SIS_ID, SOURCE_SYSTEM)  # Matching section
-            insert_edfi_section(connection, SIS_ID, SECTION_ID)  # Matching section
-
-            insert_lms_section(
-                connection, NOT_MATCHING_SIS_ID, SOURCE_SYSTEM
-            )  # Not matching section
-            insert_edfi_section(
-                connection, "also_not_matching_sis_id"
-            )  # Not matching section
-
-        # act
-        run_harmonizer(test_db_config)
-
-        # assert
-        with PgsqlConnection(test_db_config).pyodbc_conn() as connection:
-            LMSSection = query(
-                connection,
-                "select edfisectionid, sissectionidentifier from lms.lmssection",
-            )
-            assert len(LMSSection) == 2
-            assert LMSSection[0]["sissectionidentifier"] == SIS_ID
-            assert LMSSection[0]["edfisectionid"] == SECTION_ID
-            assert LMSSection[1]["sissectionidentifier"] == NOT_MATCHING_SIS_ID
-            assert LMSSection[1]["edfisectionid"] is None
+def describe_when_harmonizing_canvas_sections():
```

Comment:
I now regret adding this new level of test name, because it is harder to review. What did I actually change?

* each identifier is unique, see for example lines 28 and 53 below.
* select statements are specific, knowing that there may be other records hanging out.